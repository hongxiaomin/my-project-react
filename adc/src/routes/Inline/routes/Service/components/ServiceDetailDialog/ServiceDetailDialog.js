import React from 'react'
import { connect } from 'react-redux'
import classes from './ServiceDetailDialog.scss'
import DialogFilter from '../DialogFilter'
import OurToaster from '../../../../components/OurToaster'

type Props = {
  isOpen: boolean,
  DialogID: Number,
  DialogName: String,
  modelId: Number,
  kernelId: Number,
  toggleDialog: Function,
  getModel: Function,
  getServiceDetail: Function,
  that: Object,
  modelList: Array
};
export class ServiceDetailDialog extends React.Component {
  constructor(props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this)
    this.cb = this.cb.bind(this)
    this.state = {
      name: undefined,
      modelId: undefined,
      isChanged: false,
      kernelIds: [],
    }
  }

  componentDidMount() {
    // this.props.getModel()
  }

  handleSubmit() {
    const tmp = {
      name: this.state.name || this.props.DialogName,
      modelId: this.state.modelId || this.props.modelId,
      kernelId: this.props.kernelId,
      dataSource: '/image/T00B351BP/',
      description: 'Test',
      id: this.props.DialogID,
    }
    this.props.getServiceDetail(tmp, this.cb)
  }

  cb(data) {
    if (data.code === 200000) {
      this.props.that.isEdit = true
      this.props.toggleDialog()
      OurToaster.show({ message: '更新成功！' })
    } else {
      OurToaster.show({ message: '更新失败！' })
    }
  }

  props: Props;

  render() {
    return (
      <div className={classes['ServiceDetailDialog-container']}>
        <DialogFilter
          DialogTitle='Service Detail'
          DialogID={this.props.DialogID}
          modelId={(this.state.modelId === undefined && this.props.modelId) || this.state.modelId}
          DialogName={this.props.DialogName}
          DialogBtnContent='Edit'
          isOpen={this.props.isOpen}
          toggleDialog={this.props.toggleDialog}
          handleSubmit={this.handleSubmit}
          handleChange={this.handleChange}
          getModel={this.props.getModel}
          isChanged={this.state.isChanged}
          that={this}
          modelList={this.props.modelList}
          kernelId={this.props.kernelId}
          kernelIds={[this.props.kernelId]}
        />
      </div>
    )
  }
}

export default connect(
  state => ({
    modelList: state.service.modelList,
  }),
)(ServiceDetailDialog)
