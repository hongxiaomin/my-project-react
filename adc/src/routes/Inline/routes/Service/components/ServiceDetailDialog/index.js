import ServiceDetailDialog from './ServiceDetailDialog'

export default ServiceDetailDialog
