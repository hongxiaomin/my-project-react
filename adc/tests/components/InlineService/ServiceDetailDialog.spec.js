import React from 'react'
import { ServiceDetailDialog } from 'routes/Inline/routes/Service/components/ServiceDetailDialog/ServiceDetailDialog'

describe('(Component) ServiceDetailDialog', () => {
  const that = {}
  const DialogID = 1
  const modelId = 1
  const kernelId = 1
  const DialogName = 'a'
  const modelList = []
  const isOpen = false
  const toggleDialog = jest.fn()
  const getModel = jest.fn()
  const getServiceDetail = jest.fn()
  const dispatch = jest.fn()
  it('render normally', () => {
    const wrapper = shallow(
      <ServiceDetailDialog
        that={that}
        DialogID={DialogID}
        modelId={modelId}
        kernelId={kernelId}
        DialogName={DialogName}
        modelList={modelList}
        getModel={getModel}
        getServiceDetail={getServiceDetail}
        isOpen={isOpen}
        toggleDialog={toggleDialog}
        dispatch={dispatch}
      />,
    )
    wrapper.find('Button').last().simulate('click')
    expect(wrapper).toMatchSnapshot()
  })
})
