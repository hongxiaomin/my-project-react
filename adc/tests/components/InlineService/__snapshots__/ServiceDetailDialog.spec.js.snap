// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ServiceDetailDialog render normally 1`] = `
<div
  className="ServiceDetailDialog-container"
>
  <Connect(DialogFilter)
    DialogBtnContent="Edit"
    DialogID={1}
    DialogName="a"
    DialogTitle="Service Detail"
    getModel={[Function]}
    handleSubmit={[Function]}
    isChanged={false}
    isOpen={false}
    kernelId={1}
    kernelIds={
      Array [
        1,
      ]
    }
    modelId={1}
    modelList={Array []}
    that={
      ServiceDetailDialog {
        "_reactInternalInstance": ShallowComponentWrapper {
          "_calledComponentWillUnmount": false,
          "_compositeType": 0,
          "_context": Object {},
          "_currentElement": <ServiceDetailDialog
            DialogID={1}
            DialogName="a"
            dispatch={[Function]}
            getModel={[Function]}
            getServiceDetail={[Function]}
            isOpen={false}
            kernelId={1}
            modelId={1}
            modelList={Array []}
            that={Object {}}
            toggleDialog={[Function]}
      />,
          "_debugID": 3,
          "_hostContainerInfo": null,
          "_hostParent": null,
          "_instance": [Circular],
          "_mountOrder": 1,
          "_pendingCallbacks": null,
          "_pendingElement": null,
          "_pendingForceUpdate": false,
          "_pendingReplaceState": false,
          "_pendingStateQueue": null,
          "_renderedComponent": NoopInternalComponent {
            "_currentElement": <div
              className="ServiceDetailDialog-container"
      >
              <Connect(DialogFilter)
                      DialogBtnContent="Edit"
                      DialogID={1}
                      DialogName="a"
                      DialogTitle="Service Detail"
                      getModel={[Function]}
                      handleChange={undefined}
                      handleSubmit={[Function]}
                      isChanged={false}
                      isOpen={false}
                      kernelId={1}
                      kernelIds={
                              Array [
                                      1,
                                    ]
                      }
                      modelId={1}
                      modelList={Array []}
                      that={[Circular]}
                      toggleDialog={[Function]}
              />
      </div>,
            "_debugID": 4,
            "_renderedOutput": <div
              className="ServiceDetailDialog-container"
      >
              <Connect(DialogFilter)
                      DialogBtnContent="Edit"
                      DialogID={1}
                      DialogName="a"
                      DialogTitle="Service Detail"
                      getModel={[Function]}
                      handleChange={undefined}
                      handleSubmit={[Function]}
                      isChanged={false}
                      isOpen={false}
                      kernelId={1}
                      kernelIds={
                              Array [
                                      1,
                                    ]
                      }
                      modelId={1}
                      modelList={Array []}
                      that={[Circular]}
                      toggleDialog={[Function]}
              />
      </div>,
          },
          "_renderedNodeType": 0,
          "_rootNodeID": 0,
          "_topLevelWrapper": null,
          "_updateBatchNumber": null,
          "_warnedAboutRefsInRender": false,
        },
        "cb": [Function],
        "context": Object {},
        "handleSubmit": [Function],
        "props": Object {
          "DialogID": 1,
          "DialogName": "a",
          "dispatch": [Function],
          "getModel": [Function],
          "getServiceDetail": [Function],
          "isOpen": false,
          "kernelId": 1,
          "modelId": 1,
          "modelList": Array [],
          "that": Object {},
          "toggleDialog": [Function],
        },
        "refs": Object {},
        "state": Object {
          "isChanged": false,
          "kernelIds": Array [],
          "modelId": undefined,
          "name": undefined,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      }
    }
    toggleDialog={[Function]}
  />
</div>
`;
